articles/django-crud-api/index.md

```diff
@@ -0,0 +1,293 @@
+$ python3 manage.py startapp todo
+```
+The above command creates a directory `todo` and generates boilerplate code for a todo app. Now we can plug our todo app into our `django_todo` project.
+
+Open `settings.py` in the project directory `django_todo` and add our `todo` app to the `INSTALLED_APPS` list. We also add the `rest_framework` app to make it available for use in our project.
+```python
+# ./django_todo/settings.py
+...
+INSTALLED_APPS = [
+    ...
+    'django.contrib.messages',
+    'django.contrib.staticfiles',
+    'todo',
+    'rest_framework',
+]
+```
+### Creating the Todo model
+
+Models are python classes that get mapped to the database tables in the database. The Todo Model will be mapped to the todo table in the database. All Django models subclass the model. Model superclass.
+In the `models.py` in the `todo` app, we will create our `Todo` model will the below code.
+
+```python
+from django.db import models
+
+# Create your models here.
+class Todo(models.Model):
+    title = models.CharField(max_length = 100)
+    body = models.CharField(max_length = 100)
+    is_completed = models.BooleanField(default=False)
+    date_created = models.DateField(auto_created=True)
+    last_modified = models.DateField(auto_now=True)
+
+    def __str___(self):
+        return self.title 
+```
+To create an SQLite database and Todo table in the database, we will run the command `./manage.py makemigrations` to create SQL queries that will be used to create a `Todo` table in the database. To create the Todo table in the database will run the command `./manage.py migrate`.
+
+### Creating the TodoSerializer
+
+For converting the python objects we get from the database into a JSON format which we uses for our endpoints and conversion of JSON to python object which can be mapped to our database tables we will be subclassing a Django rest framework `Serializer.ModelSerializer` class for easier conversion.
+In the todo app directory let's create a file `serializers.py` where we will write our `TodoSerializer` code.
+
+```python
+from rest_framework import serializers
+from todo.models import Todo
+
+class TodoSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Todo
+        fields = "__all__"
+```
+
+We only need to pass in the model that we need to serialize and the fields to serialize as meta fields. In our case, we are serializing all of the fields from the `Todo` model that's why we are passing `__all__` to our `fields` variable.
+
+### Creating the TodoAPIView
+In the `todo` app directory in the `views.py` file, we need to write in the logic for the CRUD functionality for our app. Django Rest framework comes with inbuilt classes which make building the CRUD functionality very easy. We start by importing the Django rest framework classes that we will use to create our CRUD API. We just need to create a class for each of our crud endpoints and add in the `queryset` and the `serializer_class` for our `Todo` model.
```

Comment:
Consider rearranging this to:
"In the `views.py` file in our `todo` app directory, we are going to write the logic for the CRUD functionality for our app."